server/src/main/java/keywhiz/service/resources/automation/v2/ClientResource.java

```diff
@@ -98,7 +98,7 @@ public class ClientResource {
   @Consumes(APPLICATION_JSON)
   public Response createClient(@Auth AutomationClient automationClient,
       @Valid CreateClientRequestV2 request) {
-    permissionCheck.checkAllowedOrThrow(automationClient, Action.CREATE);
+    permissionCheck.checkAllowedForTargetTypeOrThrow(automationClient, Action.CREATE, Client.class);
 
```

Comment:
All of the two-argument isAllowed calls now use the "ForTargetType" method instead and supply a third argument. We already had a follow-up task for this work, but the need to implement our Secret creation permission check just moved it forward in the schedule.